src/backend/distributed/commands/multi_copy.c

```diff
@@ -3427,6 +3430,11 @@ InitializeCopyShardState(CopyShardState *shardState,
 		ereport(ERROR, (errmsg("could not connect to any active placements")));
 	}
 
+	if (hasRemoteCopy)
```

Comment:
What is the difference between the remote vs local copy?

Reply:
Local copy does not use a separate connection. We can allow that in certain scenarios (e.g. triggers that insert .. select into a co-located table), although we should maybe disallow local tasks in the regular case as well.